nevergrad/optimization/optimizerlib.py

```diff
@@ -2561,3 +2561,44 @@ def recommend(self) -> p.Parameter:
 @registry.register
 class NGOpt(NGOpt10):
     pass
+
+
+# Variants dedicated to multiobjective optimization by multiple monoobjective optimization.
+@registry.register
+class NGOpt10_16(CM):
+    """This code applies 16 copies of NGOpt10 with random weights for the different objective functions."""
+
+    num_optims = 16
+
+    def __init__(
+        self, parametrization: IntOrParameter, budget: tp.Optional[int] = None, num_workers: int = 1
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        self.optims = [
+            NGOpt10(self.parametrization, budget=1 + (budget // self.num_optims), num_workers=num_workers)
+            for _ in range(self.num_optims)
+        ]
+        self.coeffs = [np.random.uniform(size=self.num_objectives) for _ in range(self.num_optims)]
+        self.budget_before_choosing = budget
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        for i, opt in enumerate(self.optims):
+            try:
+                if self.num_objectives > 0:
```

Comment:
it's supposed to always be >0 at this point, why this condition?

Reply:
Ok.